src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -81,27 +84,31 @@ public void SetFunctionInvocationCallback(Func<Task> callback)
             }
             catch (Exception e)
             {
+                string exceptionDetails = e.ToString();
                 this.config.TraceHelper.FunctionFailed(
                     this.context.HubName,
                     this.context.Name,
                     this.context.Version,
                     this.context.InstanceId,
-                    e.ToString(),
+                    exceptionDetails,
                     FunctionType.Orchestrator,
                     this.context.IsReplaying);
+
                 if (!this.context.IsReplaying)
                 {
                     this.context.AddDeferredTask(() => this.config.LifeCycleNotificationHelper.OrchestratorFailedAsync(
                         this.context.HubName,
                         this.context.Name,
                         this.context.Version,
                         this.context.InstanceId,
-                        e.ToString(),
+                        exceptionDetails,
                         FunctionType.Orchestrator,
                         this.context.IsReplaying));
                 }
 
-                throw;
+                throw new OrchestrationFailureException(
```

Comment:
>OrchestrationFailureException [](start = 26, length = 29)

`OrchestrationFailureException` is the equivalent of `TaskFailureException` but for orchestrations and sub-orchestrations. We need to throw this exception specifically to ensure that the exception details are surfaced to the calling orchestrator function.